crates/sequencer/src/sequencer.rs

```diff
@@ -392,7 +395,23 @@ where
                 let mut txs_to_remove = self.db_provider.last_block_tx_hashes()?;
                 txs_to_remove.extend(l1_fee_failed_txs);
 
-                self.mempool.remove_transactions(txs_to_remove);
+                self.mempool.remove_transactions(txs_to_remove.clone());
+
+                match pg_pool.clone() {
+                    Some(pg_pool) => {
+                        // TODO: Is this okay? I'm not sure because we have a loop in this and I can't do async in spawn_blocking
```

Comment:
Not okay because we will have inconsistency with rocksdb and pg if a process is killed while removing txs.

Reply:
we don't put mempool txs to rocksdb btw, only written to memory

Reply:
what would you suggest not to have inconsistent pg? Is there a way to handle it better in sequencer? 
Maybe we can have a separate listener outside of sequencer and every time a block is published it removes the txs of block from pg ?